clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -305,13 +305,18 @@ public KafkaProducer(Properties properties, Serializer<K> keySerializer, Seriali
     @SuppressWarnings("unchecked")
     private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
         try {
-            log.trace("Starting the Kafka producer");
             Map<String, Object> userProvidedConfigs = config.originals();
             this.producerConfig = config;
             this.time = Time.SYSTEM;
             clientId = config.getString(ProducerConfig.CLIENT_ID_CONFIG);
             if (clientId.length() <= 0)
                 clientId = "producer-" + PRODUCER_CLIENT_ID_SEQUENCE.getAndIncrement();
+            LogContext logContext = new LogContext("[Producer clientId=" + clientId +
+                    (!userProvidedConfigs.containsKey(ProducerConfig.TRANSACTIONAL_ID_CONFIG) ? "] " :
+                            String.format(", transactionalId=%s] ", userProvidedConfigs.get(ProducerConfig.TRANSACTIONAL_ID_CONFIG))));
+            log = logContext.logger(getClass());
```

Comment:
Let's use `KafkaProducer.class` instead of `getClass()`. The logger is not exposed to sub-classes, so the context should be this class.